app/userAuthentication.js

```diff
@@ -0,0 +1,59 @@
+import fetch from "electron-fetch/lib"
+import appInfo from "./app-info.json"
+import keytar from "keytar"
+import {BrowserWindow, session} from "electron"
+
+let mainWindow, authWindow
+
+const clientId = appInfo["client_id"]
+const clientSecret = appInfo["client_secret"]
+const requiredScopes = ["repo"].join("%20")
+
+export function authorizeUser (mainWindowRef) {
+  mainWindow = mainWindowRef
+  openAuthWindow()
+}
+
+export function fetchAccessToken (code) {
+  authWindow.destroy()
+  var data = {
```

Comment:
:art: I'm not sure what your feels are on `var/let/const` but I'd enable [`no-var`](https://eslint.org/docs/rules/no-var) for the codebase at some stage